DataOnion/Auth/IAuthStorable.cs

```diff
@@ -0,0 +1,11 @@
+using StackExchange.Redis;
+
+namespace DataOnion.Auth
+{
+    public interface IAuthStorable<T>
+    {
+        string GetId();
+        bool IsSameSessionAs(T other);
```

Comment:
we should probably implement IComparable: https://learn.microsoft.com/en-us/dotnet/api/system.icomparable?view=net-6.0

Reply:
and if not that, there are other ways to do this in .NET

Reply:
I could see IEquatable, but IComparable? What does it mean for a session to be greater than (or less than) another session?

The idea is that this method is called to check whether an existing session can be re-used for the new one. For example, if we disallow the user from logging in on more than one device at a time, this would return `false` if the usernames are the same but the devices are different.

Reply:
this is what i meant by there are other .NET ways to do this, find it and use it